firefox-ios/Client/Frontend/Home/Homepage Rebuild/HomepageDiffableDataSource.swift

```diff
@@ -77,9 +90,10 @@ final class HomepageDiffableDataSource:
     private func getTopSites(
         with topSitesState: TopSitesSectionState,
         and textColor: TextColor?
-    ) -> [HomepageDiffableDataSource.HomeItem] {
-        guard topSitesState.numberOfTilesPerRow != 0 else { return [] }
+    ) -> [HomepageDiffableDataSource.HomeItem]? {
+        guard topSitesState.shouldShowSection, topSitesState.numberOfTilesPerRow != 0 else { return nil }
         let topSites: [HomeItem] = topSitesState.topSitesData.compactMap { .topSite($0, textColor) }
+        guard !topSites.isEmpty else { return nil }
         let filterTopSites = topSites.prefix(Int(topSitesState.numberOfRows) * topSitesState.numberOfTilesPerRow)
         return Array(filterTopSites)
     }
```

Comment:
@Cramsden we discussed that this may live better elsewhere, but can do this in a separate PR. Let me know if any issues or if you prefer the change in this PR!